### ISSUE TITLE
Can work without root permission

### ISSUE THREAD

**Issue by ShixiangWang** [r-lib/rig#39]
Hi,

Today I tried to install rim in a HPC without root permission. I found it require `sudo` to install R release. Could rim work for such case?

Best,

Shixiang

**Comment by gaborcsardi** [comment]
I agree that this would be great, but unfortunately it is hard, because the Linux builds need to be in `/opt`. There is no easy way around this currently.

**Comment by ShixiangWang** [comment]
Thanks. :)

Perhaps pull the R and its dependencies from conda ecosystem? I don't if this is easier. Just a thought.

**Comment by gaborcsardi** [comment]
If you use conda, then you don't need rim, you can just do everything in conda. So we'll not use conda.

**Comment by gaborcsardi** [comment]
Seems like we only need to rewrite a couple of files:
```
ubuntu@foo:/opt/R/devel$ ag /opt/R/devel
bin/R
4:R_HOME_DIR=/opt/R/devel/lib/R
5:if test "${R_HOME_DIR}" = "/opt/R/devel/lib/R"; then
19:     if [ -x "/opt/R/devel/${libnn}/R/bin/exec/R" ]; then
20:        R_HOME_DIR="/opt/R/devel/${libnn}/R"
21:     elif [ -x "/opt/R/devel/${libnn_fallback}/R/bin/exec/R" ]; then
22:        R_HOME_DIR="/opt/R/devel/${libnn_fallback}/R"
35:R_SHARE_DIR=/opt/R/devel/lib/R/share
37:R_INCLUDE_DIR=/opt/R/devel/lib/R/include
39:R_DOC_DIR=/opt/R/devel/lib/R/doc

lib/R/bin/R
4:R_HOME_DIR=/opt/R/devel/lib/R
5:if test "${R_HOME_DIR}" = "/opt/R/devel/lib/R"; then
19:     if [ -x "/opt/R/devel/${libnn}/R/bin/exec/R" ]; then
20:        R_HOME_DIR="/opt/R/devel/${libnn}/R"
21:     elif [ -x "/opt/R/devel/${libnn_fallback}/R/bin/exec/R" ]; then
22:        R_HOME_DIR="/opt/R/devel/${libnn_fallback}/R"
35:R_SHARE_DIR=/opt/R/devel/lib/R/share
37:R_INCLUDE_DIR=/opt/R/devel/lib/R/include
39:R_DOC_DIR=/opt/R/devel/lib/R/doc

lib/R/etc/Makeconf
7:# configure  '--prefix=/opt/R/devel' '--enable-R-shlib' '--with-tcltk' '--enable-memory-profiling' '--with-x' '--with-blas' '--with-lapack' 'R_PRINTCMD=/usr/bin/lpr' 'R_PAPERSIZE=letter' 'R_BROWSER=xdg-open' 'R_PDFVIEWER=xdg-open' 'LIBnn=lib'

lib/pkgconfig/libR.pc
1:rhome=/opt/R/devel/lib/R
3:rincludedir=/opt/R/devel/lib/R/include
```

**Comment by h4rvey-g** [comment]
Hi, is this feature available now?

**Comment by nathan-nguyen-sa** [comment]
Would like an update on this as well. I also need to install R releases without root permission.

**Comment by groverj3** [comment]
Hey there, would like to add that this would be a great feature for those of us working in HPC or multi-user server environments where it's desirable to have users manage their own R installations. Similar to what's possible with pyenv.

**Comment by gaborcsardi** [comment]
To clarify. 

* We could relatively easily implement the use case where admin installs various R versions, and each user can set their own default.
* It is much harder to implement the use case where users install their own R installations, because our R builds are not self-contained and they are not relocatable. I.e. they require a bunch of system packages, and they have to live in `/opt`.

**Comment by groverj3** [comment]
> To clarify.
> 
> * We could relatively easily implement the use case where admin installs various R versions, and each user can set their own default.
> * It is much harder to implement the use case where users install their own R installations, because our R builds are not self-contained and they are not relocatable. I.e. they require a bunch of system packages, and they have to live in `/opt`.

This makes sense. Of course, I respect the work that has already gone into this! Even just having the first scenario as a viable option would be a fantastic enhancement. I don't know how many people care, but there are at least a few of us.

**Comment by h4rvey-g** [comment]
For those who struggle to install R without root permission, docker would be a good option, e.g. [https://rocker-project.org/](Rocker project)

**Comment by gaborcsardi** [comment]
Consideration is one thing, and implementation is another. The main issue is not relocation (https://github.com/r-lib/rig/issues/39#issuecomment-1104959378) but that the R builds also need a bunch of system packages, which you'd need to use `sudo` to install, anyway.

**Comment by gaborcsardi** [comment]
This is not fixed with user mode rig: https://rig.r-lib.org/admin-vs-user-mode.html
